viewing_party/party.py

```diff
@@ -0,0 +1,215 @@
+#
+#Robot invasion wave 1
+#
+from typing import Counter
+from statistics import mode
+
+def create_movie(title,genre,rating):
+    if bool(title) and bool(genre) and  bool(rating):
+        movie= {"title": title, "genre":genre, "rating": rating}
+        return movie
+    else:
+        return None
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie) 
+    return user_data
+  
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data,title):
+
+    for i,dct in enumerate(user_data["watchlist"]):
+        if title == dct["title"]:
+                watched_movie_dct = user_data["watchlist"].pop(i)
+                user_data["watched"].append(watched_movie_dct)
+                return user_data
+    return user_data
+    
+#
+#Robot invasion Wave 2
+#
+def get_watched_avg_rating(user_data):
+    sum=0
+    if user_data["watched"]==[]:
+        return 0
+    
+    for dct in user_data["watched"]:
+        sum += dct["rating"]
+    return sum/len(user_data["watched"])
+
+def get_most_watched_genre(user_data):
+    most_watched_lst=[]
+
+    if user_data["watched"]==[]:
+        return None
+    for mvie in user_data["watched"]:
+        most_watched_lst.append(mvie["genre"])
+    genre_cnt = Counter(most_watched_lst)
+    top_genre= genre_cnt.most_common(1)
+    return top_genre[0][0]
+#
+# Robot invasion wave 3
+# we ran out of drinks, please send taro boba tea
+#
+def get_unique_watched(user_data):
+    listof_watched_dct=[]
+    a_lst=[]
+    b_lst=[]
+    ab_lst=[]
+    user_movie_lst=[]
+    friends_movie_lst=[]
+    new_movie_lst=[]
+    #This for loop creates a new list with all the movies that the user watched
+    for dct in user_data["watched"]:
+        user_movie_lst.append(dct)
+
+    #This for loop creates two lists of movies that friends watched
+    for dct in user_data["friends"]:
+        for lst in dct:
+            listof_watched_dct.append(dct[lst])
+    a_lst= (listof_watched_dct[0])
+    b_lst= (listof_watched_dct[1])
+    ab_lst= a_lst+b_lst
+
+    for i in ab_lst:
```

Comment:
`i` is generally used to represent a numerical value when iterating. Consider giving this variable a more meaningful name.